Detector/DetSensitive/src/SegmentationHelper.cpp

```diff
@@ -12,22 +12,25 @@
 
 namespace det {
 namespace segmentation {
-uint64_t cellID(const DD4hep::Geometry::Segmentation& aSeg, const G4Step& aStep) {
+uint64_t cellID(const DD4hep::Geometry::Segmentation& aSeg, const G4Step& aStep, bool aPreStepPoint) {
   DD4hep::Simulation::Geant4VolumeManager volMgr =
     DD4hep::Simulation::Geant4Mapping::instance().volumeManager();
   DD4hep::Geometry::VolumeManager::VolumeID volID =
     volMgr.volumeID(aStep.GetPreStepPoint()->GetTouchable());
   if (aSeg.isValid() )  {
-    G4ThreeVector global = 0.5 * (  aStep.GetPreStepPoint()->GetPosition()+
-      aStep.GetPostStepPoint()->GetPosition());
+    G4ThreeVector global;
+    if (aPreStepPoint)
+      global = aStep.GetPreStepPoint()->GetPosition();
+    else
+      global = 0.5*(aStep.GetPreStepPoint()->GetPosition()+aStep.GetPostStepPoint()->GetPosition());
```

Comment:
Could you add curly braces to avoid problems if somebody changes the code in the future?